src/script/lua_api/l_object.cpp

```diff
@@ -170,16 +170,28 @@ int ObjectRef::l_punch(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
-	ObjectRef *puncher_ref = checkObject<ObjectRef>(L, 2);
+	ObjectRef *puncher_ref = nullptr;
+	try {
+		puncher_ref = checkObject<ObjectRef>(L, 2);
+	}
+	catch (...) {
+		puncher_ref = nullptr;
+	}
 	ServerActiveObject *sao = getobject(ref);
-	ServerActiveObject *puncher = getobject(puncher_ref);
-	if (sao == nullptr || puncher == nullptr)
+	ServerActiveObject *puncher = (puncher_ref)?(getobject(puncher_ref)):(nullptr);
```

Comment:
code style: spacing and useless braces
(also applies elsewhere)
```suggestion
	ServerActiveObject *puncher = puncher_ref ? getobject(puncher_ref) : nullptr;
```

Reply:
Fixed on more lines. Hopefully, I did not miss anything.